src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -718,6 +728,12 @@ private bool AreNamedTypesEqual(NamedTypeSymbol type, NamedTypeSymbol other)
                 // TODO: Test with overloads (from PE base class?) that have modifiers.
                 Debug.Assert(!type.HasTypeArgumentsCustomModifiers);
                 Debug.Assert(!other.HasTypeArgumentsCustomModifiers);
+
+                if (type.IsTupleType)
+                {
+                    return other.IsTupleType && type.TupleElementTypes.SequenceEqual(other.TupleElementTypes, AreTypesEqual);
+                }
+
                 return type.TypeArgumentsNoUseSiteDiagnostics.SequenceEqual(other.TypeArgumentsNoUseSiteDiagnostics, AreTypesEqual);
```

Comment:
Replaced this block with asserts, as it was only there speculatively (it's protected by the other change in the symbol matcher, as far as we can tell).